File: python_programmers_school/level 1/추억 점수.py
def solution(name, yearning, photo):
    result = []
    for i in photo:
        sum = 0
        for j in i:
            if j in name:
                sum += yearning[name.index(j)]
        result.append(sum)
    return result

print(solution(["may", "kein", "kain", "radi"], [5, 10, 1, 3], [["may", "kein", "kain", "radi"],["may", "kein", "brin", "deny"], ["kon", "kain", "may", "coni"]]))

File: javascript_programmers_school/level 1/추억 점수.js
function solution(name, yearning, photo) {
  return photo.map((nameList) => {
    let score = 0;
    nameList.map((n) => {
      if (name.indexOf(n) != -1) score += yearning[name.indexOf(n)];
    });
    return score;
  });
}

// 간단한 풀이
function solution(name, yearning, photo) {
  return photo.map((v) =>
    // yearning 리스트에 name.indexOf(c)의 값이 존재하지 않는 경우(인덱스 초과) 0이 누적됨. (undefined이므로)
    v.reduce((a, c) => (a += yearning[name.indexOf(c)] ?? 0), 0)
  );
}
